### PR TITLE
(DOCSP-29224): Data class document data type 

### PR DESCRIPTION
**Pull request by mongodben**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-29224
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29224/fundamentals/data-formats/document-data-format-data-class/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/fundamentals/data-formats/document-data-format-data-class.txt [hunk 1]
```diff
@@ -15,4 +15,155 @@ Document Data Format: Data Classes
...
+In this guide, you can learn how to store and retrieve data in the
+{+driver-long+} using **Kotlin data classes**.
+
+Serialize and Deserialize a Data Class
+--------------------------------------
+
+The driver natively supports encoding and decoding Kotlin data classes for
+MongoDB read and write operations using the **default codec registry**. The
+default codec registry is a collection of classes called **codecs** that
+define how to convert encode and decode Kotlin types.
```
Comment (jyemin) [inline source/fundamentals/data-formats/document-data-format-data-class.txt:27]:
```suggestion
define how to encode and decode Kotlin and Java types.
```

File: source/fundamentals/data-formats/document-data-format-data-class.txt [hunk 2]
```diff
@@ -15,4 +15,155 @@ Document Data Format: Data Classes
...
+
+   * - Annotation Name
+     - Description
+
+   * - ``BsonId``
+     - Marks a property to serialize as the _id property.
+
+   * - ``BsonProperty``
+     - Specifies a custom document field name when converting the data class
+       field to BSON. You can include a discriminator to serialize data classes
```
Comment (jyemin) [inline source/fundamentals/data-formats/document-data-format-data-class.txt:89]: Remove the sentence about discriminators.  The data class codec (and the record codec, for that matter), ignore the `useDiscriminator` field of `BsonProperty`.

**Review by jyemin — APPROVED** [review]
Couple of small change requests, but no need for re-review after knocking them out.

File: source/fundamentals/data-formats/document-data-format-data-class.txt [hunk 3]
```diff
@@ -15,4 +15,155 @@ Document Data Format: Data Classes
...
+       nested within the field.
```
Comment (mongodben) [inline source/fundamentals/data-formats/document-data-format-data-class.txt:90]:
```suggestion
       field to BSON.
```

**Pull request merged**